src/test/ui/feature-gates/feature-gate-unboxed-closures-manual-impls.rs

```diff
@@ -8,24 +8,28 @@
 struct Foo;
 impl Fn<()> for Foo {
 //~^ ERROR the precise format of `Fn`-family traits' type parameters is subject to change
+//~| ERROR manual implementations of `Fn` are experimental
```

Comment:
It's not clear to me how the attribute checking changes made this new error to appear.

Reply:
I don't know the rustc code well yet, so I can only tell where this difference happens, not why.

Without this PR, `rustc_typeck::check_crate()` errors out early in the "type_collecting" step (I assume `tcx.ensure().collect_mod_item_types()` fails?), so the the code never makes it to the `coherence::check_coherence()` call a few lines below.

While I have no idea why my PR fixes the type_collecting, it does look like an improvement - all new error messages seem correct to me.

Reply:
We could maybe use `-Ztreat-err-as-bug` to determine why this happens? This would give us a backtrace which hopefully would include `check_attrs`.